update greenboot config in customisation

say-paul opened the pull request:
This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

<!--
Thanks for proposing a change to osbuild-composer!

Please don't remove the above check list. These are things that each pull
request must have before it is merged. It helps maintainers to not forget
anything.

If the reason for ticking any of the boxes is ambiguous, please add a short
note explaining why.

In addition, if this pull request fixes a downstream issue, please refer to
test/README.md and add these additional items:

- [ ] 1st commit of any `rhbz#` related PR contains bug reproducer; CI reports FAIL or
- [ ] PR contains automated tests for new functionality and
- [ ] QE has approved reproducer/new tests and
- [ ] Subsequent commits provide bug fixes without modifying the reproducer; CI reports PASS and
- [ ] QE approves this PR; RHBZ status is set to `MODIFIED + Verified=Tested`

Information regarding our GitLab pipeline (Schutzbot):

CI will not be ran automatically if WIP label is applied or the PR is in DRAFT state, instead only
a link will be provided to the pipeline which can then be triggered manually if desired. To run the
CI automatically either switch the PR to ready or apply WIP+test label.

Outside contributors need manual approval from one of the osbuild-composer maintainers.

Schutzbot will only be triggered if all Tests jobs in GitHub workflow succeed.
-->

lzap committed:
dbjobqueue: use background context when closing listener

diaasami committed:
appsre-ansible: support aarch64

make ansible playbooks arch-agnostic
extract embedded bash script into separate file with parameters
update packer template to support aarch64
Convert parts of bash script to python code that can start multi-arch instances to build RPMS

lzap committed:
Introduce logging adapter for jobqueue

croissanne committed:
worker: Define new jobs to handle copying and resharing of images

The copy job copies from one region to another. It does not preserve the
sharing on the ami and it's snapshot, that needs to be queued
separately.

croissanne committed:
cloudapi: Add endpoints to clone aws images across regions

Support for creating multiple amis from a single compose. It uses the
AWSEC2* jobs to push images to new regions, and share them with new
accounts.

The compose it depends upon has to have succeeded.

lzap committed:
Add IntelliJ IDEA/GoLand to .gitignore

croissanne committed:
jobqueue: Query job dependents

croissanne committed:
worker: Return dependent jobs in OsbuildJobStatus

diaasami committed:
appsre-ansible: subscribe before upgrading packages

F-X64 committed:
internal/osbuild: Add GCP guest-agent conf stage

Add GCP guest-agent conf stage test

F-X64 committed:
internal/distro/rhel8: Add GCPGuestAgentConf

Add NewGcpGuestAgentConfStage to osPipeline

F-X64 committed:
internal/distro/rhel90: Add GCPGuestAgentConf

Add NewGcpGuestAgentConfStage to osPipeline

F-X64 committed:
internal/distro: Add GCPGuestAgentConf to ImageConfig

F-X64 committed:
test/data/manifests: regenerate test cases RHEL+CentOS 8,9

F-X64 committed:
osbuild-composer.spec: osbuild version bump 64

Schutzfile: Bump osbuild commit hash to include new GCP guest-agent stage

juan-abia committed:
test: disable non supported tests

some scripts skip the test if it's not supported for that
distro-version. Disable them in gitlab-ci.yml so we don't waste CI
resources.

To disable them, we are using the `rules` on each job with a regex
pattern. Using `=~` (pattern matches) as a  WHITELIST and `!~` (pattern
does not match) as a BLACKLIST.

atodorov committed:
Enable integration testing on aarch64

NOTE: yaml can merge maps but not sequences

kingsleyzissou committed:
distro: add oscap packages to image

Since the oscap remediation stage in osbuild runs
the oscap package in `chroot`, it is necessary to
install the `openscap-scanner` package to the image
itself rather than the build root.

brlane-rht committed:
dnf-json: Add search command

Use the DNF query API
(https://dnf.readthedocs.io/en/latest/api_queries.html) to quickly
return results matching a glob pattern. Multiple package glob results
are combined into a single response.

This adds a search dict to the arguments. 'packages' is a list of package
names or globs to search for.
An optional 'latest' boolean will return only the latest NEVRA instead
of all matching builds in the metadata.

eg.

    "search": {
        "latest": false,
        "packages": ["tmux", "vim*", "*ssh*"]
    },

brlane-rht committed:
dnfjson: Add the search support to the Solver

Pass the list of package names or globs to dnf-json and return the
results.

brlane-rht committed:
weldr: Use the SearchMetadata function for globs

Instead of fetching all available packages from dnf-json and then
searching the results this uses SearchMetadata when a package name or
glob is passed to the API. It only uses FetchMetadata when fetching
the full list of packages.

This also fixes a bug where the error response to a projects/info
request used the id of 'ModulesError'. It now uses 'ProjectsError'.

brlane-rht committed:
mocks/dnfjson: Add mock support for the search command

The search command is more complicated than depsolve and dump. It needs
to return results based on the requested package names and globs.

Add a number of mock responses for the new search command, including
search results, all packages, and error responses that are triggered by
using special package names: nonexistingpkg, badpackage1, baddepsolve.

runcom commented on internal/blueprint/customizations.go:
```diff
@@ -23,6 +23,11 @@ type Customizations struct {
+}
+
+type GreenbootCustomization struct {
+	MonitorService string `json:"monitor_service,omitempty" toml:"monitor_service,omitempty"`
```
should be `MonitorServices` plural, and why a string? and not an array

runcom commented on internal/blueprint/customizations.go:
```diff
@@ -23,6 +23,11 @@ type Customizations struct {
 	InstallationDevice string                    `json:"installation_device,omitempty" toml:"installation_device,omitempty"`
 	FDO                *FDOCustomization         `json:"fdo,omitempty" toml:"fdo,omitempty"`
 	OpenSCAP           *OpenSCAPCustomization    `json:"openscap,omitempty" toml:"openscap,omitempty"`
+	Greenboot          *GreenbootCustomization   `json:"greenboot" toml:"greenboot,omitempty"`
```
call it GreenbootConfig if you're writing GetGreenbootConfig

runcom commented on internal/osbuild/greenboot_stage.go:
```diff
@@ -0,0 +1,14 @@
+package osbuild
+
+type GreenbootConfig struct {
+	MonitorService string `json:"monitor_services,omitempty"`
```
the stage in https://github.com/osbuild/osbuild/pull/1086 does define this as an array of strings so that here too

say-paul committed:
greenboot monitor service:
This enables user to specify the list of services greenboot
 require monitoring from
blueprint.
signed-off:paul.sayan@gmail.com

added greenboot stage test

changes include:
* variable name change to prural
* added support for rhel8

adding default services for greenboot monitoring.

changes include:
* stage options for greenboot
* moving green from installer to ospipeline
*some variable name change

Sync greenboot stage options for rhel8 with rhel9

refactor pipeline for code optimization
greenboot stage options are assigned default values and overriden based
on the input recived.

optimizing greenboot config with default:
removed stage options for adding things into pipeline which make
the code simple and staright forward.

moved greenboot config to rpmostree block
moving greenboot config to rpmostree ensures the its added  only to the
rpmostree build

fix greenboot stage

Signed-off-by: Antonio Murdaca <runcom@linux.com>

require osbuild >= 65

Signed-off-by: Antonio Murdaca <runcom@linux.com>

jrusz committed:
Update terraform SHA with more aarch64 runner options

This commit adds the option to also provision c6g.large machines in
addition to c7g.large machines in AWS.

brlane-rht committed:
api_test: Use badpackage1 request for FetchError request

This returns the FetchError response setup in mocks/dnfjson for the
search command.

brlane-rht committed:
mock-dnf-json: Add support for search command

The search response from mocks/dnfjson is a map of responses indexed by the
comma-separated list of packages and globs being requested. Add support
for this.

schutzbot committed:
Post release version bump

[skip ci]

thozza committed:
Drop special handling for RHEL < 8.4 in `GetHostDistroName()`

RHEL-8.3 is EOL and we do not explicitly support it. There is no need to
mangle the host distro name for RHEL-8 releases older than 8.4.

thozza committed:
Simplify trimming of quotes in `readOSRelease()`

Use standard library function to trim surrounding quotes from read
values. This makes the code shorter and easier to read.

achilleas-k committed:
distro/rhel9: enable hybrid boot for edge-installer

Regression introduced when unifying distro versions.

achilleas-k committed:
test: update test manifests (RHEL 9.x and CS9)

achilleas-k committed:
distro/rhel8: enable hybrid boot for edge-installer

Regression introduced when unifying distro versions.

achilleas-k committed:
test: update test manifests (RHEL 8.x and CS8)

henrywang committed:
test: Add edge-install BISO installation test back for BZ#2110864

thozza committed:
worker/osbuild: use `os-release` to determine host OS

When running an osbuild job, we read `/etc/redhat-release` to get the
host OS name to attach as metadata to the job result.

Only Fedora and RHEL ship this file, which makes the osbuild job always
fail on other distributions.

The main reason to report host OS back to the worker server is due to
Koji composes and the koji-finalize job, which pushes it to Koji. The
motivation is to have enough information to potentially re-instantiate
/ identify the original builder host OS. There are no specific
requirements on the string.

Modify the code to use `/etc/os-release` to determine the host OS. Fall
back to using `linux` as the host OS, in case reading `os-release`
fails, log the error and continue with the job. The `linux` fallback is
suggested by the `os-release` spec [1]

[1] https://www.freedesktop.org/software/systemd/man/os-release.html#ID=

Co-authored-by: Achilleas Koutsou <achilleas@koutsou.net>

achilleas-k committed:
gen-manifests: don't fail on invalid distro-arch-image combos

When the user specifies any of the distro, arch, or image type values to
filter generation, invalid combinations would cause a panic, which made
it hard to filter requests based just on an image type.
Instead of failing, print an error message to inform the user, but
continue with the rest of the jobs.
This way, a user is informed that a certain combination is invalid if
they make a mistake, but can also filter on a single image type and only
get valid manifests out of the run.

achilleas-k committed:
distro/fedora: enable hybrid boot for iot-installer

achilleas-k committed:
test: update test manifests (Fedora 35 and 36)

supakeen committed:
Add the `rhsm.facts` stage.

We initially provide a Facts file that stores the `ApiType`. This is the
API that was used to request the compose.

jrusz committed:
test/gcp: Run cleanup function at the end

All EXIT traps are cleared on line 280 so the cleanup trap is never run
and VMs are waiting for 4 hours to get cleaned by
scheduled-cloud-cleaner. Run the cleanup at the end and rely on
scheduled-cloud-cleaner only in case of failures before that.

croissanne committed:
cloudapi/v2: Add rhsm facts for non-koji cloudapi composes

croissanne committed:
distro: Only add rhsm facts stage for RHEL

croissanne committed:
cloudapi/v2: Don't add rhsm facts

There are workers everywhere which can't deal with this change just
yet. Let's wait until workers can deal with this before adding facts via
the api.

croissanne committed:
schutzbot: Fast-forward release branch after green main run

The release branch can be tracked by stage deployments to make sure CI
is green before deploying to stage.

schutzbot committed:
Post release version bump

[skip ci]

ondrejbudai committed:
go.mod: update github.com/containers/image/v5

Version 5.22 introduced a new option to /etc/containers/policy.json called
keyPaths, see

https://github.com/containers/image/pull/1609

EL9 immediately took advantage of this new feature and started using it, see
https://gitlab.com/redhat/centos-stream/rpms/containers-common/-/commit/04645c4a84442da3324eea8f6538a5768e69919a

This quickly became an issue in our code: The go library (containers/image)
parses the configuration file very strictly and refuses to create a client
when policy.json with an unknown key is present on the filesystem. As we
used 5.21.1 that doesn't know the new key, our unit tests started to
failing when containers-common was present.

Reproducer:
podman run --pull=always --rm -it centos:stream9
dnf install -y dnf-plugins-core
dnf config-manager --set-enabled crb
dnf install -y gpgme-devel libassuan-devel krb5-devel golang git-core
git clone https://github.com/osbuild/osbuild-composer
cd osbuild-composer

# install the new containers-common and run the test
dnf install -y https://kojihub.stream.centos.org/kojifiles/packages/containers-common/1/44.el9/x86_64/containers-common-1-44.el9.x86_64.rpm
go test -count 1 ./...

# this returns:
--- FAIL: TestClientResolve (0.00s)
    client_test.go:31:
        	Error Trace:	client_test.go:31
        	Error:      	Received unexpected error:
        	            	Unknown key "keyPaths"
        	            	invalid policy in "/etc/containers/policy.json"
        	            	github.com/containers/image/v5/signature.NewPolicyFromFile
        	            		/osbuild-composer/vendor/github.com/containers/image/v5/signature/policy_config.go:88
        	            	github.com/osbuild/osbuild-composer/internal/container.NewClient
        	            		/osbuild-composer/internal/container/client.go:123
        	            	github.com/osbuild/osbuild-composer/internal/container_test.TestClientResolve
        	            		/osbuild-composer/internal/container/client_test.go:29
        	            	testing.tRunner
        	            		/usr/lib/golang/src/testing/testing.go:1439
        	            	runtime.goexit
        	            		/usr/lib/golang/src/runtime/asm_amd64.s:1571
        	Test:       	TestClientResolve
    client_test.go:32:
        	Error Trace:	client_test.go:32
        	Error:      	Expected value not to be nil.
        	Test:       	TestClientResolve

 When run with an older containers-common, it succeeds:
 dnf install -y https://kojihub.stream.centos.org/kojifiles/packages/containers-common/1/40.el9/x86_64/containers-common-1-40.el9.x86_64.rpm
 go test -count 1 ./...
 PASS

To sum it up, I had to upgrade github.com/containers/image/v5 to v5.22.0.
Unfortunately, this wasn't so simple, see

go get github.com/containers/image/v5@latest
go: github.com/containers/image/v5@v5.22.0 requires
	github.com/letsencrypt/boulder@v0.0.0-20220331220046-b23ab962616e requires
	github.com/honeycombio/beeline-go@v1.1.1 requires
	github.com/gobuffalo/pop/v5@v5.3.1 requires
	github.com/mattn/go-sqlite3@v2.0.3+incompatible: reading github.com/mattn/go-sqlite3/go.mod at revision v2.0.3: unknown revision v2.0.3

It turns out that github.com/mattn/go-sqlite3@v2.0.3+incompatible has been
recently retracted https://github.com/mattn/go-sqlite3/pull/998 and this
broke a ton of packages depending on it. I was able to fix it by adding

exclude github.com/mattn/go-sqlite3 v2.0.3+incompatible

to our go.mod, see
https://github.com/mattn/go-sqlite3/issues/975#issuecomment-955661657

After adding it,
go get github.com/containers/image/v5@latest
succeeded and tools/prepare-source.sh took care of the rest.

Signed-off-by: Ondřej Budai <ondrej@budai.cz>

ondrejbudai committed:
blueprint: remove omitempty from Customizations

See https://github.com/BurntSushi/toml/issues/360

A recent change in BurntSushi/toml made encoding fail (later changed to
error) if a struct is marked as omitempty and is comparable. Go docs about
equality: https://go.dev/doc/go1#equality. Basically: A struct is comparable
if all of its fields are comparable. Slices are not comparable.

Customizations are marked as omitempty but they contain a lot of slices,
thus they are not comparable. The new version of BurntSushi/toml therefore
panics when we encode them.

The solution is to remove the omitempty tag from Customizations.

Signed-off-by: Ondřej Budai <ondrej@budai.cz>

diaasami committed:
.gitlab-ci: add rhel-8.6-ga-aarch64 runner for AWS API tests

diaasami committed:
api/common/aws.sh: support arm64 when installing govc

schutzbot committed:
Post release version bump

[skip ci]

jrusz committed:
CI: update test execution on nightly pipelines

We don't support regression-bigiso and koji tests in nightly pipelines
so don't run them.

dependabot[bot] committed:
build(deps): bump github.com/jackc/pgx/v4 from 4.16.0 to 4.17.1

Bumps [github.com/jackc/pgx/v4](https://github.com/jackc/pgx) from 4.16.0 to 4.17.1.
- [Release notes](https://github.com/jackc/pgx/releases)
- [Changelog](https://github.com/jackc/pgx/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jackc/pgx/compare/v4.16.0...v4.17.1)

---
updated-dependencies:
- dependency-name: github.com/jackc/pgx/v4
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

juan-abia committed:
test: get latest version of CIV in aws.sh

instead of manually updating CIV version every once in a while. Get
always the latest version.

In CIV CI, this test runs before any change can be introduced into the
container image, so no unexpedted errors should come from the CIV side.

diaasami committed:
test/cases/api/aws.sh: add support for starting aarch64 instances

brlane-rht committed:
weldr: Preload metadata at startup

For each of the supported distros start a goroutine to depsolve
'filesystem' which will preload the metadata making subsequent responses
faster.

This is safe to do without limits because we only supposed a limited
number of distros, and without additional locking because this is the
the same as hitting the API with multiple depsolve requests at the same
time.

dependabot[bot] committed:
build(deps): bump github.com/containers/common from 0.48.0 to 0.49.1

Bumps [github.com/containers/common](https://github.com/containers/common) from 0.48.0 to 0.49.1.
- [Release notes](https://github.com/containers/common/releases)
- [Commits](https://github.com/containers/common/compare/v0.48.0...v0.49.1)

---
updated-dependencies:
- dependency-name: github.com/containers/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/prometheus/client_golang

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.12.1 to 1.13.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.12.1...v1.13.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump cloud.google.com/go/storage from 1.22.1 to 1.26.0

Bumps [cloud.google.com/go/storage](https://github.com/googleapis/google-cloud-go) from 1.22.1 to 1.26.0.
- [Release notes](https://github.com/googleapis/google-cloud-go/releases)
- [Changelog](https://github.com/googleapis/google-cloud-go/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-cloud-go/compare/pubsub/v1.22.1...spanner/v1.26.0)

---
updated-dependencies:
- dependency-name: cloud.google.com/go/storage
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/Azure/go-autorest/autorest

Bumps [github.com/Azure/go-autorest/autorest](https://github.com/Azure/go-autorest) from 0.11.27 to 0.11.28.
- [Release notes](https://github.com/Azure/go-autorest/releases)
- [Changelog](https://github.com/Azure/go-autorest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/Azure/go-autorest/compare/autorest/v0.11.27...autorest/v0.11.28)

---
updated-dependencies:
- dependency-name: github.com/Azure/go-autorest/autorest
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/golang-jwt/jwt/v4 from 4.4.1 to 4.4.2

Bumps [github.com/golang-jwt/jwt/v4](https://github.com/golang-jwt/jwt) from 4.4.1 to 4.4.2.
- [Release notes](https://github.com/golang-jwt/jwt/releases)
- [Changelog](https://github.com/golang-jwt/jwt/blob/main/VERSION_HISTORY.md)
- [Commits](https://github.com/golang-jwt/jwt/compare/v4.4.1...v4.4.2)

---
updated-dependencies:
- dependency-name: github.com/golang-jwt/jwt/v4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/gophercloud/gophercloud

Bumps [github.com/gophercloud/gophercloud](https://github.com/gophercloud/gophercloud) from 0.24.0 to 1.0.0.
- [Release notes](https://github.com/gophercloud/gophercloud/releases)
- [Changelog](https://github.com/gophercloud/gophercloud/blob/master/CHANGELOG.md)
- [Commits](https://github.com/gophercloud/gophercloud/compare/v0.24.0...v1.0.0)

---
updated-dependencies:
- dependency-name: github.com/gophercloud/gophercloud
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/labstack/echo/v4 from 4.7.2 to 4.8.0

Bumps [github.com/labstack/echo/v4](https://github.com/labstack/echo) from 4.7.2 to 4.8.0.
- [Release notes](https://github.com/labstack/echo/releases)
- [Changelog](https://github.com/labstack/echo/blob/master/CHANGELOG.md)
- [Commits](https://github.com/labstack/echo/compare/v4.7.2...v4.8.0)

---
updated-dependencies:
- dependency-name: github.com/labstack/echo/v4
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

atodorov committed:
tests: Terraform RPM available only on x86_64

according to
https://www.hashicorp.com/blog/announcing-the-hashicorp-linux-repository
https://discuss.hashicorp.com/t/rhel-8-aarch64-rpms/24244/3

and
https://github.com/hashicorp/packer/issues/11516

Don't execute azure.sh if the runner is not x86_64

diaasami committed:
tests: add aarch64 rhel-9.0 runner to API tests

atodorov committed:
tests: Execute vmware testing only on x86_64

achilleas-k committed:
rhel9: explicitly add containernetworking-plugins to edge

In podman v4.0.0 the default network backend was switched from cni to
netavark.  However, podman will choose cni if there are already
containers, images, or cni networks preset on a system [1].

Starting with podman v4.2.0, containernetworking-plugins is no longer a
hard requirement for podman.  So when an edge commit is built with an
embedded container, podman v4.2.0+ will choose the cni network and fail
with an error because the plugin isn't installed.

Adding the package explicitly alongside podman to avoid this issue with
future RHEL 9.1 edge builds when they include containers.

This change does not affect test manifests.  The package is already
included in manifests as a dependency of podman < v4.2.0.

See rhbz#2123210

[1] https://github.com/containers/common/blob/a083f790abf21048518f2d66b7e01370cc055790/pkg/config/containers.conf#L275-L278

thozza committed:
distro: use pointers for basic types in `ImageConfig`

Using basic types as values in the `ImageConfig` structure makes it
impossible to distinguish if the empty value for the type was set
intentionally or if it is just the value the variable was initialized
to. This is very bad especially for `bool` type.

While working on unifying `vhd` and `azure-rhui` image types I found
out, that some newly added variables in the `ImageConfig` structure
were forgotten in the `InheritFrom()` method. This makes it impossible
to inherit their values from a parent configuration. This is however
required for the unification of `vhd` and `azure-rhui` image types. As
described above, it would be impossible to decide whether a `bool` value
should be inherited from the parent configuration or not. The only
solution is to use a pointer to the type. For consistency, use pointer
for all basic types.

Adjust distro implementations accordingly.

thozza committed:
distro: use reflection in `ImageConfig.InheritFrom()`

As it turned out, people make mistakes and forget to write some parts of
code, unless a unit test screams at them. This is true for the
`InheritFrom()` method, which is not handling all members of the
`ImageConfig` structure.

Use reflection, instead of inheriting from each specific hard-coded
structure member. This will make the implementation future-proof in case
the `ImageConfig` structure is extended with additional members.

croissanne committed:
templates/packer: Increase aws timeouts for rhel-8-aarch64

This job is failing with "ResourceNotReady: exceeded wait attempts".

https://www.packer.io/plugins/builders/amazon#resourcenotready-error

juan-abia committed:
test: change CIV tag

We were using `latest` as tag, this can be dangerous as it's the default
tag, an anyone can accidentally update it. Using `prod` is safer.

Also use dev container image if the test script is running in CIV CI.

schutzbot committed:
schutzfile: Update snapshots to 20220906

Updates all snapshots bar centos-stream-9, as tests using libvirt are
broken with the newer snapshots.

schutzbot committed:
Post release version bump

[skip ci]

diaasami committed:
appsre: disable aarch64 AMI creation until issue is resolved

after merging of PR #2718, generation of AMIs has been failing with 'ResourceNotReady: exceeded wait attempts'.
issue tracked in #2961

runcom committed:
regerate manifests

Signed-off-by: Antonio Murdaca <runcom@linux.com>

dependabot[bot] committed:
build(deps): bump github.com/aws/aws-sdk-go from 1.44.44 to 1.44.93

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.44.44 to 1.44.93.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Changelog](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.44.44...v1.44.93)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

atodorov committed:
Adjust AWS EC2 type in tests depending on CPU arch

runcom committed:
update snapshots

Signed-off-by: Antonio Murdaca <runcom@linux.com>

runcom committed:
move greenboot stage before prep stage

Signed-off-by: Antonio Murdaca <runcom@linux.com>

diaasami committed:
templates/packer: reasonable aws_polling limits for rhel AWS builds

diaasami committed:
tools/appsre: disable packer parallel builds

diaasami committed:
Revert "appsre: disable aarch64 AMI creation until issue is resolved"

This reverts commit 84f46eebdb3e988a88d8813d921d2a4064411159.

runcom committed:
regenerate manifests

Signed-off-by: Antonio Murdaca <runcom@linux.com>

jrusz committed:
test/upgrade: install vdo package

This package is required for the upgrade process.

jrusz committed:
CI: Introduce x86_64 rules

We have recently added testing on aarch64 but not all of the test are
supported on that architecture. Adding specific rules for all and x86_64
arches to simplify which test jobs run on what arches.
